docs/notary.md

```diff
@@ -0,0 +1,531 @@
+performed by the notary service, do not require user's intervention and is given
+for informational purposes. Stage 4 contains tips to check notary request
+results.
+
+### 1. Notary deposit
+
+Notary service is not free and is paid through the sender's deposit that
+contained in the Notary native contract. Before the notary request will be
+submitted, you need to deposit enough GAS to the contract, otherwise, request
+won't pass verification.
+
+Notary native contract supports `onNEP17Payment` method, thus to deposit funds to
+the Notary native contract, transfer desired amount of GAS to the contract
+address. Use
+[func (*Client) TransferNEP17](https://pkg.go.dev/github.com/nspcc-dev/neo-go@v0.97.2/pkg/rpc/client#Client.TransferNEP17)
+with the `data` parameter matching the following requirements:
+- `data` should be an array of two elements: `to` and `till`.
+- `to` denotes the receiver of the deposit. It can be nil in case if `to` equals
+  to the GAS sender.
+- `till` denotes chain's height before which deposit is locked and can't be
+  withdrawn. `till` can't be set if you're not the deposit owner. Default `till`
+  value is current chain height + 5760. `till` can't be less than current chain
+  height. `till` can't be less than previously set value.
+
+Note, that first deposit call for the `to` address can't be less than 2×FEE GAS.
+Deposit is allowed for renewal, i.e. consequent `deposit` calls for the same `to`
+address add up specified amount to the already deposited value.
+
+After GAS transfer successfully submitted to the chain, use [Notary native
+contract API](#Native Notary contract) to manage your deposit.
+
+Note, that regular operation flow requires deposited amount of GAS to be
+sufficient to pay for the fallback transaction fees which are system fee and
+network fee. Fallback network fee includes (`NKeys`+1)×`FEE` = (0+1)×`FEE` = `FEE`
+GAS for `NotaryAssisted` attribute usaage and regular fee for the fallback size.
+If the deposited amount is not enough to pay the fallback fees, then
+`Insufficiend funds` error will be returned from the RPC node after notary
+request submission.
+
+### 2. Request submission
+
+Once several parties want to sign one transaction, each of them should generate
+the transaction, wrap it into `P2PNotaryRequest` payload and send to the known RPC
+server via [`submitnotaryrequest` RPC call](./rpc.md#submitnotaryrequest-call).
+
+To create notary request, you can use [NeoGo RPC client](./rpc.md#Client). Follow
+the steps to create a signature request:
+
+1. Prepare list of signers with scopes for the main transaction (i.e. the
+   transaction that signatures are being collected for, that will be `Signers`
+   transaction field). Use the following rules to construct the list:
+   * First signer is the one who pays transaction fees.
+   * Each signer is either multisignature or standard signature or a contract
+     signer.
+   * Multisignature and signature signers can't be combined.
```

Comment:
Is it a limitation of our CLI or the whole notary service?

Reply:
Notary service. It can be removed in future, but for now it simplifies things a bit.

This can be rephrased like `incomplete multisignature and single-signature signers`, because transactions can have complete signatures of any kind.

Reply:
The current implementation does not support combining complete multisignature with single-signature witnesses, see `(n *Notary) verifyIncompleteWitnesses` method. Also, invocation script of incomplete multisignature witness is allowed to contain only **one** signature.

@roman-khimov should we adjust these two things?

Reply:
That's more about the case where multisig is accompanied by a number of standard sigs, that is possible. Other than that we're not changing Notary logic now, we're just describing it. We can submit some issue for future work though, some limitations are rather obvious and maybe we'll remove them in future.

Reply:
> multisig is accompanied by a number of standard sigs, that is possible

Current implementation does not allow this, I added an issue (https://github.com/nspcc-dev/neo-go/issues/2150). Documentation comment remains untouched for now:
```
 * Multisignature and signature signers can't be combined.
```